test/system/trader.js

```diff
@@ -1,65 +1,42 @@
-const  CryptoniteClient = require('../../lib/cryptonite-client');
+const CryptoniteClient = require('../../lib/cryptonite-client');
 const logInitialDetails = require('./initilaLogger');
 
+let orderbook = {
+  prices: [],
+  quantities: []
+};
+
+let traders = {};
+
 class Trader {
 
   constructor(config) {
     this.config = config;
     this.client = new CryptoniteClient(this.config);
     this.market = 'LTC/BTC';
-    this.init_state(this.config);
-  }
-
-  init_state() {
-    return new Promise((resolve)=> {
-      this._myorders = {updated: undefined, data: undefined, errors: []};
-      this.client.connect()
-        .then(() => {
-          this.client.getBalances()
-            .then(data => {
-              const balanceData = data.balances;
-              this.config.balance.assets = balanceData.btc.availableBalance;
-              this.config.balance.capital = balanceData.ltc.availableBalance;
-              this.config.state.start.assets = balanceData.btc.availableBalance;
-              this.config.state.start.capital = balanceData.ltc.availableBalance;
-              this.logInitialDetails(this.config);
-              return this.client.getMyOrders(1, 4);
-            })
-            .then((data) => {
-              this._myorders.updated = new Date();
-              this._myorders.data = data.orders;
-              console.log('current orderd are : ', this._myorders);
-              this.start_trading();
-              resolve();
-            })
-            .catch(err =>{console.log(err);});
-        });
-    });
-
-  }
-
-  logInitialDetails(config) {
-    logInitialDetails({
-      config,
-    });
+    this.client.connect()
```

Comment:
Since constructors cannot be async, anything that involves promises that you need to wait for will need to be in a member function (like init_state)